tools/buildsys/src/manifest.rs

```diff
@@ -347,6 +361,42 @@ impl ManifestInfo {
             .as_ref()
             .and_then(|m| m.build_variant.as_ref())
     }
+
+    #[allow(clippy::collapsible_match)]
+    fn collect_paths(
+        &self,
+        root_path: &PathBuf,
+        dependencies: &HashMap<String, DependencyInfo>,
+        paths: &mut HashSet<String>,
+    ) {
+        for info in dependencies.values() {
+            if let DependencyInfo::Dependency { path } = info {
+                if let Some(p) = path {
+                    if p != "." {
+                        let mut path = p.clone();
+                        // Handle relative paths and things like storewolf:merge-toml
```

Comment:
This? Assuming we expect all of these paths to exist: https://doc.rust-lang.org/std/fs/fn.canonicalize.html

Reply:
Just make the canonicalize call right up front and use that to determine path maybe?

Reply:
Sorry, I was thinking canonicalize would make the path absolute, but that's not correct. There is this though: https://docs.rs/path-absolutize/latest/path_absolutize/

Reply:
Things work well with the current use of canonicalize, so I'm still not really sure what your suggestion is here. I do think it could make sense to just make that call and check its result rather than having a substring check first, then doing it. I can change that in the next update if we decide to proceed. But want to make sure I'm not missing something else here.

Reply:
I've changed this section. If there is something you were pointing out that I totally missed, please let me know.